openpype/plugins/publish/extract_burnin.py

```diff
@@ -690,7 +690,7 @@ def prepare_repre_data(self, instance, repre, burnin_data, temp_data):
         burnin_data.update({
             "frame_start": burnin_frame_start,
             "frame_end": burnin_frame_end,
-            "duration": burnin_duration,
+            "duration": burnin_duration
```

Comment:
En theorie d'ailleurs, la duration sera toujours egale au burnin_frame_end
Que l'on soit en keep_frame_index ou non

Reply:
C'est là où on peut pas faire ça car oui peut-être que ça marche pour Tv Paint mais pour les autres softs, la duration est le nombre de frame entre start et end, pas la frame end, exemple:
```
burnin_frame_start = 12
burnin_frame_end = 21
```
Là normalement la duration c'est `burnin_frame_end - burnin_frame_start + 1`, c'est le frame_count

Reply:
A voir comment il reagit dans les autres softs en effet.
Dans ce cas il faut mettre en place le set de cette valeur ou l'ajout d'une variable pour le otio_burnin modulo dans if is_sequence

Reply:
Hello @hfarre, j'ai regardé le problème, il s'avère que la commande exécuté dans le otio_burnin : ```C:\Users\ccaillot\quad\OpenPype\vendor\bin\ffmpeg\windows\bin\ffmpeg.exe -start_number 17 -i "C:/Users/ccaillot/AppData/Local/Temp/tvpaint_render_hxmxsc6t/_jpg/_jpg.%02d.jpg" -filter_script:v "C:\Users\ccaillot\AppData\Local\Temp\tmpzhd8hx0h" -start_number 17 -codec:v mjpeg -pix_fmt yuvj420p -g 1 -y "C:/Users/ccaillot/AppData/Local/Temp/tvpaint_render_hxmxsc6t/_jpg/_jpg.%02dburnin_tvpaint.jpg"``` génére la séquence suivante : _jpg.17burnin_tvpaint.jpg, _jpg.18burnin_tvpaint.jpg, _jpg.19burnin_tvpaint.jpg.

Après quelques tests, je me suis rendu compte qu'en enlevant le deuxième flag `start_number`, on rendait une séquence compatible au résultat attendu par OP de telle sorte à avoir cette commande : 
```C:\Users\ccaillot\quad\OpenPype\vendor\bin\ffmpeg\windows\bin\ffmpeg.exe -start_number 17 -i "C:/Users/ccaillot/AppData/Local/Temp/tvpaint_render_hxmxsc6t/_jpg/_jpg.%02d.jpg" -filter_script:v "C:\Users\ccaillot\AppData\Local\Temp\tmpzhd8hx0h" -codec:v mjpeg -pix_fmt yuvj420p -g 1 -y "C:/Users/ccaillot/AppData/Local/Temp/tvpaint_render_hxmxsc6t/_jpg/_jpg.%02dburnin_tvpaint.jpg"``` générant la séquence suivante : _jpg.01burnin_tvpaint.jpg, _jpg.02burnin_tvpaint.jpg, _jpg.03burnin_tvpaint.jpg.

Le problème étant que l'ajout de ce flag est forcé par le burnin (cf [here](https://github.com/quadproduction/OpenPype/blob/1b82670cbe455b3e2ff968cd46eb79bbd54d061f/openpype/scripts/otio_burnin.py#L450)). Je suis en train de voir ce que je peux faire de mon côté.

Reply:
@hfarre petite update, j'ai résolu le problème de mon côté. Il me reste certains points à traiter cependant.

Premièrement, pour résoudre le problème, j'ai étudié la logique du burnin dans Openpype, j'ai testé a travers plusieurs softs, et j'ai remarqué qu'a chaque fois, on lui envoie toujours en input un .mp4. Alors que dans notre cas, on lui envoie une sequence d'image, alors que l'on peut trés bien générer un .mp4 a partir cet séquence d'image sans soucis vu qu'on génère nous mêmes les images qui ne sont pas définit par le user ("fill the gap"), on est d'accord ? 

A partir de cet logique, je me suis rendu compte que dans le extract_sequence.py, on en venait à spécifier cela ```custom_tags.append("sequence")``` et dans ton commentaire tu as mis ``` # It makes no sense to make a video of discontinuous frame range```.  Et c'est sur ce point qu'arrive l'erreur, car l'on peut très bien générer une vidéo à partir de notre séquence d'images ?

De plus, j'ai vu que tu as lié ce `custom_tag` au profil `jpg_single` et que tu l'as setté sur 'Only if input has 1 image frame'.

Je pense qu'il y a confusion dans la logique car oui dans les faits, on sélectionne des images a publié mais dans la matière, on remplit les trous pour en faire une séquence ? Donc le profil `jpg_single` n'a pas lieu d'être et ainsi on peut utilisé le profil `h264` qui s'appuira sur notre séquence afin de créer le burnin comme la logique d'OP le souhaite.

J'ai besoin de faire des tests complémentaires afin de m'assurer que tout fonctionne correctement dans l'ensemble.

Reply:
@hfarre petite update, j'ai résolu le problème de mon côté. Il me reste certains points à traiter cependant.

Premièrement, pour résoudre le problème, j'ai étudié la logique du burnin dans Openpype, j'ai testé a travers plusieurs softs, et j'ai remarqué qu'a chaque fois, on lui envoie toujours en input un .mp4. Alors que dans notre cas, on lui envoie une sequence d'image, alors que l'on peut trés bien générer un .mp4 a partir de cette séquence d'image sans soucis vu qu'on génère nous mêmes les images qui ne sont pas définit par le user ("fill the gap"), on est d'accord ?

A partir de cet logique, je me suis rendu compte que dans le extract_sequence.py, on en venait à spécifier cela custom_tags.append("sequence") et dans ton commentaire tu as mis  # It makes no sense to make a video of discontinuous frame range. Et c'est sur ce point qu'arrive l'erreur, car l'on peut très bien générer une vidéo à partir de notre séquence d'images ?

De plus, j'ai vu que tu as lié ce custom_tag au profil jpg_single et que tu l'as setté sur 'Only if input has 1 image frame'.

Je pense qu'il y a confusion dans la logique car oui dans les faits, on sélectionne des images a publié mais dans la matière, on remplit les trous pour en faire une séquence ? Donc le profil jpg_single n'a pas lieu d'être et ainsi on peut utilisé le profil h264 qui s'appuira sur notre séquence afin de créer le burnin comme la logique d'OP le souhaite.

Cet ajout palie également au fait qu'on peut transmettre au burnin une séquence qui ne se suit pas, cependant, je pense qu'il est nécessaire de faire le tri des files en amont via le repre['files'], et également dans le cas on l'on a un export_frames de ce type [16, 18, 20], il faut setter la duration à 3, start_frame à 16, end_frame à 20, qu'en penses etu ? 

J'ai besoin de faire des tests complémentaires afin de m'assurer que tout fonctionne correctement dans l'ensemble.

Reply:
Le check ne se fait pas lorsqu'il s'agit d'un .mp4 en entrée, vu que le formatage de la duration sur l'output ne peut pas sa faire.

Reply:
Exact, le soucis c'est que nous on veut lui faire faire une séquence d'img qui peut ne pas correspondre en terme d'index à la même valeur que la duration
Dans [16, 18, 20] la dernière frame est d'index 20 et non 3 celle de la duration